crates/uv/src/commands/venv.rs

```diff
@@ -142,6 +144,29 @@ async fn venv_impl(
     if preview.is_enabled() && interpreter_request.is_none() {
         interpreter_request = request_from_version_file().await.into_diagnostic()?;
     }
+    if preview.is_enabled() && interpreter_request.is_none() {
+        let project = match VirtualProject::discover(
+            &std::env::current_dir().into_diagnostic()?,
+            &DiscoveryOptions::default(),
+        )
+        .await
+        {
+            Ok(project) => Some(project),
+            Err(WorkspaceError::MissingPyprojectToml) => None,
+            Err(WorkspaceError::NonWorkspace(_)) => None,
+            Err(err) => return Err(err).into_diagnostic(),
+        };
+
+        if let Some(project) = project {
+            interpreter_request = find_requires_python(project.workspace())
+                .into_diagnostic()?
+                .as_ref()
+                .map(RequiresPython::specifiers)
+                .map(|specifiers| {
+                    PythonRequest::Version(VersionRequest::Range(specifiers.clone()))
+                });
+        }
```

Comment:
Code here is adopted from `run.rs`:
https://github.com/astral-sh/uv/blob/c0d3da8b6ae134036b3ed024762b08c4fb9f967b/crates/uv/src/commands/project/run.rs#L176-L182
and `FoundInterpreter` in `mod.rs`:
https://github.com/astral-sh/uv/blob/c0d3da8b6ae134036b3ed024762b08c4fb9f967b/crates/uv/src/commands/project/mod.rs#L158-L172